services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +42,52 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if (&thermalMgrQueueHandle != NULL) {
+    if (event != NULL) {
+      if (xQueueSend(thermalMgrQueueHandle, (void *) event, (TickType_t) 0) == pdPASS)
+        return ERR_CODE_SUCCESS;
+      else
+        return ERR_CODE_QUEUE_FULL;
+    }
+    else
+      return ERR_CODE_INVALID_ARG;
+  }
 
-  return ERR_CODE_SUCCESS;
+  return ERR_CODE_INVALID_QUEUE_MSG;
 }
```

Comment:
Also note that when your doing multiple errror checks like this it's a lot cleaner to just use multiple if statements with early returns rather than a bunch of nested if statements. For example something like ```

if(errCodeCondition){ 
  return ERR_CODE
}
if(otherPossibleError){
  return OTHER_ERROR
}
// continue with the rest of the function knowing everything before was success without need an else statement
```